### PR TITLE
Speed up Trainer Initialization

### PR DESCRIPTION
**Pull request by ramppdev**
This outsources the following I/O related tasks during startup to separate threads to speed up the trainer:
- saving the `requirements.txt`
- generating and saving the model summary
- saving the initial (and initial) best states for the model, optimizer, and scheduler
- saving all audobject YAML files like the model, the target transform, ...

This was a long needed improvement and i measured ~4-6x speedups of the `__init__` function.

### CODE REVIEW

**Review by millinma — APPROVED** [review]
Seems good, if you wanna have a more in-depth review maybe wait for @ATriantafyllopoulos.

File: autrainer/training/training.py [hunk 1]
```diff
@@ -211,33 +220,18 @@ def __init__(
...
-        )
-        self.bookkeeping.save_audobject(self.model, "model.yaml")
-        self.bookkeeping.save_audobject(
-            self.data.test_transform, "inference_transform.yaml"
-        )
-        self.bookkeeping.save_audobject(
-            self.data.file_handler, "file_handler.yaml"
-        )
+        # ? Save initial (and best) Model, Optimizer and Scheduler states
+        [
```
Comment (ATriantafyllopoulos) [inline autrainer/training/training.py:224]: Won't it be easier to understand if you turn the list comprehension in a simple for loop?

File: autrainer/training/training.py [hunk 2]
```diff
@@ -261,12 +255,17 @@ def __init__(
                 ]
             )
 
-        self.bookkeeping.save_audobject(
-            _preprocess_pipe, "preprocess_pipeline.yaml"
-        )
-        self.bookkeeping.save_audobject(
-            _file_handler, "preprocess_file_handler.yaml"
-        )
+        [
```
Comment (ATriantafyllopoulos) [inline autrainer/training/training.py:258]: See comment above for list comprehension

File: autrainer/core/utils/hardware.py [hunk 3]
```diff
@@ -1,13 +1,26 @@
...
 import torch
 
 
+def spawn_thread(target: Callable, args: tuple = (), daemon=False) -> None:
+    """Spawn and start a new thread for the target function.
+
+    Args:
+        target: Function to run in the thread.
+        args: Arguments to pass to the function. Defaults to an empty tuple.
+        daemon: Whether the thread should be run as a daemon thread.
```
[the remark this answers is not in the document]
Comment (ramppdev) [reply autrainer/core/utils/hardware.py:18]: Not really. I first thought i would need it but can be safely removed.

**Comment by ramppdev** [comment]
@ATriantafyllopoulos all comments are addressed :)
